app/auth/defra-id/get-authorization-url.js

```diff
@@ -1,23 +1,24 @@
 const { authConfig } = require('../../config')
 const { createState } = require('./create-state')
+const { createInitialisationVector } = require('./create-initialisation-vector')
 const { getWellKnown } = require('./get-well-known')
 
-const getAuthorizationUrl = async (redirect) => {
+const getAuthorizationUrl = async (request, redirect) => {
   const { authorization_endpoint: url } = await getWellKnown()
 
-  const state = createState(redirect)
+  const state = createState(request, redirect)
+  const initialisationVector = createInitialisationVector(request)
 
-  // TODO: setup nonce
   const query = [
     `p=${authConfig.policy}`,
     `client_id=${authConfig.clientId}`,
     `serviceId=${authConfig.serviceId}`,
     `state=${state}`,
-    'nonce=defaultNonce',
+    `nonce=${initialisationVector}`,
     `redirect_uri=${authConfig.redirectUrl}`,
     `scope=openid offline_access ${authConfig.clientId}`,
     'response_type=code',
-    'response_mode=form_post'
+    'response_mode=query'
```

Comment:
Need to change to `query` as we've now changed the response route to `GET` rather than `POST`.   Reason below.